contracts/interchain-token/InterchainToken.sol

```diff
@@ -15,10 +16,15 @@ import { Minter } from '../utils/Minter.sol';
  * @title InterchainToken
  * @notice This contract implements an interchain token which extends InterchainToken functionality.
  * @dev This contract also inherits Minter and Implementation logic.
+ * Uses assembly to directly write to slot 0, bypassing parent contract storage layout.
  */
 contract InterchainToken is InterchainTokenStandard, ERC20, ERC20Permit, Minter, IInterchainToken {
```

Comment:
We need to maintain both the standard and new InterchainToken for Hyperliquid. Instead of forking the repo and having to maintain release processes for that, create a new HyperliquidInterchainToken that inherits from this contract and adds necessary modifications